src/ng/anchorScroll.js

```diff
@@ -232,8 +236,9 @@ function $AnchorScrollProvider() {
       }
     }
 
-    function scroll() {
-      var hash = $location.hash(), elm;
+    function scroll(hash) {
+      hash = (arguments.length && isString(hash)) ? hash : $location.hash();
```

Comment:
That would be the _reasonable_ thing to do :)
I was trying to avoid accidentally breaking an existing app, by ensuring that the `hash` argument is a string.
It's been a while, so I am not really sure about the `arguments.length` check; I suspect it was me micro-optimizing (or at least believing so).

It seems that `isString(hash) ? hash : $location.hash()` would be enough.

--
I would rather use `isString()` instead of `isDefined()`, to avoid breaking some unlikely construct like:

    elem.on('click', $anchorScroll);   // e.g. passing and event object will be ignored